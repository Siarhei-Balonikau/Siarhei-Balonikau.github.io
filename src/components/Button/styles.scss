.button {
  position: relative;
  border: none;
  background-color: transparent;
  display: inline-block;
  padding: 15px 40px;
  cursor: pointer; 
  
  &__inner {
    &:before {
      position: absolute; 
      content: '';
      left: 0;
      bottom: 0;
      width: 2px;
      height: 50%;
      background-color: black;
      transition: 0.3s;
    }
    
    &:after {
      position: absolute; 
      content: '';
      left: 0;
      bottom: 0;
      width: 30%;
      height: 2px;
      background-color: black;
      transition: 0.3s;
    }
  }
  
  &:before {
    position: absolute; 
    content: '';
    right: 0;
    top: 0;
    width: 2px;
    height: 50%;
    background-color: black;
    transition: 0.3s;
  }
  
  &:after {
    position: absolute; 
    content: '';
    right: 0;
    top: 0;
    width: 30%;
    height: 2px;
    background-color: black;
    transition: 0.3s;
  }
  
  &:hover {
    &:before {
      height: 100%;
    }
    
    &:after {
      width: 100%;
    }
    
    .button {
      &__inner {
        &:before {
          height: 100%;
        }
        
        &:after {
          width: 100%;
        }
      }
    }
  }
}