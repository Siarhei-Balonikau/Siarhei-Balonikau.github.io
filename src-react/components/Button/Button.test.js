import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Button from './Button'; 
import toJson from 'enzyme-to-json'

describe('subscribe component', () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      
    };
  
    wrapper = shallow(<Button {...props} />);
  });
  
  it('should render component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});