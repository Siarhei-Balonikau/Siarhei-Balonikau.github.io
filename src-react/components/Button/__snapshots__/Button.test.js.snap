// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component should render component 1`] = `
<button
  className="button"
>
  <span
    className="button__inner"
  />
</button>
`;
